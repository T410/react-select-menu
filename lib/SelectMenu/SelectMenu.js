function SelectMenu() {
	return (
		<div>
			<p1>Menu</p1>
		</div>
	);
}

export default SelectMenu;
