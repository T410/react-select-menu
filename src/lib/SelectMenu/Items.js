import { useState, useEffect } from "react";
import { sliceArray } from "../helper";
function Items({ options }) {
	// const [slicedOptions, setSlicedOptions] = useState([]);
	// useEffect(() => {
	// 	setSlicedOptions(sliceArray(options, "groupID"));
	// }, [options]);
	// return slicedOptions.map((options) => {
	// 	return options.items.map((option, index) => {
	// 		return (
	// 			<div className={index === activeOptionIndex ? activeStyles : passiveStyles} key={index}>
	// 				<div className={style.item} onClick={() => itemClickHandler({ index, option })}>
	// 					<Check />
	// 					<p>{option.name}</p>
	// 				</div>
	// 			</div>
	// 		);
	// 	});
	// });
}

export default Items;
