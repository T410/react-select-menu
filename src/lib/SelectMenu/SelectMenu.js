function SelectMenu() {
	return (
		<div>
			<p>Menu</p>
		</div>
	);
}

export default SelectMenu;
