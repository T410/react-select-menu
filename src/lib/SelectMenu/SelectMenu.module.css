.outerContainer {
	font-weight: var(--medium);
	width: fit-content;
}

.select {
	min-width: 100px;
	width: fit-content;
	height: 32px;
	box-shadow: var(--box-shadow);
	border: 1px solid var(--color-border);
	border-radius: var(--border-radius);
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: center;
}

.visibleOption {
	margin: 0 var(--s-15);
}

.select:hover {
	cursor: pointer;
}

.dropdown {
	width: fit-content;
	position: absolute;
	margin-top: var(--s-5);
	box-shadow: var(--box-shadow);
	border: 1px solid var(--color-border);
	border-radius: var(--border-radius);
}

.items {
	height: var(--s-30);
	display: flex;
	flex-direction: row;
	align-items: center;
	cursor: default;
}

.items:hover {
	background-color: var(--color-passive-hover);
}
