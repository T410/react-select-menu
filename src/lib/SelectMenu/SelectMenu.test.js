import "jest-enzyme";
import { configure, render, shallow, mount } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
configure({ adapter: new Adapter() });
import React from "react";
import SelectMenu from "./SelectMenu";
const options = [
	{
		name: "Hopper",
		value: "hopper",
		description: "Grace Hopper was an American computer scientist and US Navy rear admiral.",
		groupID: 2,
	},
	{
		name: "Holberton",
		value: "holberton",
		description: "Frances Elizabeth Holberton was one of the programmers of the first computer.",
		groupID: 2,
	},
	{
		name: "Teitelbaum",
		value: "teitelbaum",
		description: "Ruth Teitelbaum was one of the first computer programmers in the world",
		groupID: 1,
	},
	{
		name: "Tayyib",
		value: "tayyib",
		description: "Hey, it's me!",
	},
	{
		name: "No Description",
		value: "noDesc",
		groupID: 3,
	},
];
describe("SelectMenu render and functionality", () => {
	it("mounts SelectMenu withoput any props", () => {
		const wrapper = mount(<SelectMenu />);
		expect(wrapper).toMatchSnapshot();
	});

	it("renders SelectMenu with props", () => {
		const wrapper = mount(
			<SelectMenu options={options} defaultValue={"tayyib"} isSimple={false} darkMode={true} maxWidth={300} />
		);
		expect(wrapper).toMatchSnapshot();
	});

	it("shows detailed dropdown", () => {
		const wrapper = mount(
			<SelectMenu options={options} defaultValue={"tayyib"} isSimple={false} darkMode={true} maxWidth={300} />
		);
		wrapper.find("div").at(1).simulate("click");
		expect(wrapper).toMatchSnapshot();
	});

	it("calls onChange callback", () => {
		const expectedResult = {
			name: "Teitelbaum",
			value: "teitelbaum",
			description: "Ruth Teitelbaum was one of the first computer programmers in the world",
			groupID: 1,
		};
		let testingResult = {};
		const onChangeHandler = (e) => {
			testingResult = e;
		};
		const wrapper = mount(
			<SelectMenu
				options={options}
				defaultValue={"tayyib"}
				isSimple={false}
				darkMode={true}
				maxWidth={300}
				onChange={onChangeHandler}
			/>
		);
		wrapper.find("div").at(1).simulate("click");
		wrapper.find("DetailedOption").at(0).simulate("click");
		expect(testingResult).toEqual(expectedResult);
	});
});
