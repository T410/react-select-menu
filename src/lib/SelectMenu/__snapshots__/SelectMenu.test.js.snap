// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectMenu render and functionality mounts SelectMenu withoput any props 1`] = `
<SelectMenu>
  <div
    className="lightMode selectOuterContainer"
  >
    <div
      className="selectMiddleContainer"
      onClick={[Function]}
    >
      <p
        className="selectInnerContainer"
      >
        No Option
      </p>
    </div>
  </div>
</SelectMenu>
`;

exports[`SelectMenu render and functionality renders SelectMenu with props 1`] = `
<SelectMenu
  darkMode={true}
  defaultValue="tayyib"
  isSimple={false}
  maxWidth={300}
  options={
    Array [
      Object {
        "description": "Grace Hopper was an American computer scientist and US Navy rear admiral.",
        "groupID": 2,
        "name": "Hopper",
        "value": "hopper",
      },
      Object {
        "description": "Frances Elizabeth Holberton was one of the programmers of the first computer.",
        "groupID": 2,
        "name": "Holberton",
        "value": "holberton",
      },
      Object {
        "description": "Ruth Teitelbaum was one of the first computer programmers in the world",
        "groupID": 1,
        "name": "Teitelbaum",
        "value": "teitelbaum",
      },
      Object {
        "description": "Hey, it's me!",
        "name": "Tayyib",
        "value": "tayyib",
      },
      Object {
        "groupID": 3,
        "name": "No Description",
        "value": "noDesc",
      },
    ]
  }
>
  <div
    className="darkMode selectOuterContainer"
  >
    <div
      className="selectMiddleContainer"
      onClick={[Function]}
    >
      <p
        className="selectInnerContainer"
      >
        Tayyib
      </p>
    </div>
  </div>
</SelectMenu>
`;

exports[`SelectMenu render and functionality shows detailed dropdown 1`] = `
<SelectMenu
  darkMode={true}
  defaultValue="tayyib"
  isSimple={false}
  maxWidth={300}
  options={
    Array [
      Object {
        "description": "Grace Hopper was an American computer scientist and US Navy rear admiral.",
        "groupID": 2,
        "name": "Hopper",
        "value": "hopper",
      },
      Object {
        "description": "Frances Elizabeth Holberton was one of the programmers of the first computer.",
        "groupID": 2,
        "name": "Holberton",
        "value": "holberton",
      },
      Object {
        "description": "Ruth Teitelbaum was one of the first computer programmers in the world",
        "groupID": 1,
        "name": "Teitelbaum",
        "value": "teitelbaum",
      },
      Object {
        "description": "Hey, it's me!",
        "name": "Tayyib",
        "value": "tayyib",
      },
      Object {
        "groupID": 3,
        "name": "No Description",
        "value": "noDesc",
      },
    ]
  }
>
  <div
    className="darkMode selectOuterContainer"
  >
    <div
      className="selectMiddleContainer"
      onClick={[Function]}
    >
      <p
        className="selectInnerContainer"
      >
        Tayyib
      </p>
    </div>
    <div
      className="dropdown"
      style={
        Object {
          "maxWidth": 300,
        }
      }
    >
      <Option
        className="passiveOuterItem"
        isSimple={false}
        onClick={[Function]}
        option={
          Object {
            "description": "Ruth Teitelbaum was one of the first computer programmers in the world",
            "groupID": 1,
            "name": "Teitelbaum",
            "value": "teitelbaum",
          }
        }
      >
        <DetailedOption
          className="passiveOuterItem"
          onClick={[Function]}
          option={
            Object {
              "description": "Ruth Teitelbaum was one of the first computer programmers in the world",
              "groupID": 1,
              "name": "Teitelbaum",
              "value": "teitelbaum",
            }
          }
        >
          <div
            className="detailedItem passiveOuterItem"
            onClick={[Function]}
          >
            <div
              className="itemInner"
            >
              <div
                className="detailedItemInner"
              >
                <p>
                  Teitelbaum
                </p>
                <p>
                  Ruth Teitelbaum was one of the first computer programmers in the world
                </p>
              </div>
            </div>
          </div>
        </DetailedOption>
      </Option>
      <Divider>
        <div
          className="divider"
        />
      </Divider>
      <Option
        className="passiveOuterItem"
        isSimple={false}
        onClick={[Function]}
        option={
          Object {
            "description": "Grace Hopper was an American computer scientist and US Navy rear admiral.",
            "groupID": 2,
            "name": "Hopper",
            "value": "hopper",
          }
        }
      >
        <DetailedOption
          className="passiveOuterItem"
          onClick={[Function]}
          option={
            Object {
              "description": "Grace Hopper was an American computer scientist and US Navy rear admiral.",
              "groupID": 2,
              "name": "Hopper",
              "value": "hopper",
            }
          }
        >
          <div
            className="detailedItem passiveOuterItem"
            onClick={[Function]}
          >
            <div
              className="itemInner"
            >
              <div
                className="detailedItemInner"
              >
                <p>
                  Hopper
                </p>
                <p>
                  Grace Hopper was an American computer scientist and US Navy rear admiral.
                </p>
              </div>
            </div>
          </div>
        </DetailedOption>
      </Option>
      <Option
        className="passiveOuterItem"
        isSimple={false}
        onClick={[Function]}
        option={
          Object {
            "description": "Frances Elizabeth Holberton was one of the programmers of the first computer.",
            "groupID": 2,
            "name": "Holberton",
            "value": "holberton",
          }
        }
      >
        <DetailedOption
          className="passiveOuterItem"
          onClick={[Function]}
          option={
            Object {
              "description": "Frances Elizabeth Holberton was one of the programmers of the first computer.",
              "groupID": 2,
              "name": "Holberton",
              "value": "holberton",
            }
          }
        >
          <div
            className="detailedItem passiveOuterItem"
            onClick={[Function]}
          >
            <div
              className="itemInner"
            >
              <div
                className="detailedItemInner"
              >
                <p>
                  Holberton
                </p>
                <p>
                  Frances Elizabeth Holberton was one of the programmers of the first computer.
                </p>
              </div>
            </div>
          </div>
        </DetailedOption>
      </Option>
      <Divider>
        <div
          className="divider"
        />
      </Divider>
      <Option
        className="passiveOuterItem"
        isSimple={false}
        onClick={[Function]}
        option={
          Object {
            "groupID": 3,
            "name": "No Description",
            "value": "noDesc",
          }
        }
      >
        <DetailedOption
          className="passiveOuterItem"
          onClick={[Function]}
          option={
            Object {
              "groupID": 3,
              "name": "No Description",
              "value": "noDesc",
            }
          }
        >
          <div
            className="detailedItem passiveOuterItem"
            onClick={[Function]}
          >
            <div
              className="itemInner"
            >
              <div
                className="detailedItemInner"
              >
                <p>
                  No Description
                </p>
                <p />
              </div>
            </div>
          </div>
        </DetailedOption>
      </Option>
      <Divider>
        <div
          className="divider"
        />
      </Divider>
      <Option
        className="activeOuterItem itemActive"
        isSimple={false}
        onClick={[Function]}
        option={
          Object {
            "description": "Hey, it's me!",
            "groupID": Infinity,
            "name": "Tayyib",
            "value": "tayyib",
          }
        }
      >
        <DetailedOption
          className="activeOuterItem itemActive"
          onClick={[Function]}
          option={
            Object {
              "description": "Hey, it's me!",
              "groupID": Infinity,
              "name": "Tayyib",
              "value": "tayyib",
            }
          }
        >
          <div
            className="detailedItem activeOuterItem itemActive"
            onClick={[Function]}
          >
            <div
              className="itemInner"
            >
              <div
                className="detailedItemInner"
              >
                <p>
                  Tayyib
                </p>
                <p>
                  Hey, it's me!
                </p>
              </div>
            </div>
          </div>
        </DetailedOption>
      </Option>
      <Divider>
        <div
          className="divider"
        />
      </Divider>
    </div>
  </div>
</SelectMenu>
`;
